### PR TITLE
Make ServiceCollectionExtensions consistent

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

**Comment by javiercn** [comment]
Things to pay attention to:
* Comments (There's been a lot of copy paste, so mistakes can happen).
* Method names
* Null checks for arguments
* Overload with configuration action calls parameterless overload.
* Parameterless overload does the work and brings in direct depenendencies
  (Like routing on MVC, for most repositories it means making sure AddOptions is called).

File: src/Microsoft.AspNetCore.Diagnostics.Elm/ElmExtensions.cs [hunk 1]
```diff
@@ -23,9 +23,9 @@ public static IApplicationBuilder UseElmCapture(this IApplicationBuilder app)
 
             // add the elm provider to the factory here so the logger can start capturing logs immediately
             var factory = app.ApplicationServices.GetRequiredService<ILoggerFactory>();
-            var store = app.ApplicationServices.GetRequiredService<ElmStore>();
+            var provider = app.ApplicationServices.GetRequiredService<ElmLoggerProvider>();
             var options = app.ApplicationServices.GetService<IOptions<ElmOptions>>();
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Diagnostics.Elm/ElmExtensions.cs]: Unnecessary line?

**Comment by HaoK** [comment]
I would hold off on this change, `IOptions<T>` is going to be deleted real soon...

https://github.com/aspnet/Options/issues/122

**Comment by javiercn** [comment]
@pranavkm You mind taking a look at this?

File: src/Microsoft.AspNetCore.Diagnostics.Elm/ElmServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -3,45 +3,50 @@
...
-            return services;
+
+            services.AddOptions();
+            services.TryAddSingleton<ElmStore>();
+            services.TryAddSingleton<ElmLoggerProvider>();
         }
 
         /// <summary>
-        /// Configures a set of <see cref="ElmOptions"/> for the application.
+        /// Adds MVC services to the specified <see cref="IServiceCollection" />.
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Diagnostics.Elm/ElmServiceCollectionExtensions.cs]: > Adds Elm services ...

File: src/Microsoft.AspNetCore.Diagnostics.Elm/ElmServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -3,45 +3,50 @@
...
         /// </summary>
-        /// <param name="services">The services available in the application.</param>
-        /// <param name="configureOptions">The <see cref="ElmOptions"/> which need to be configured.</param>
-        public static void ConfigureElm(
-            this IServiceCollection services,
-            Action<ElmOptions> configureOptions)
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <param name="setupAction">An action delegate to configure the provided <see cref="ElmOptions"/>.</param>
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Diagnostics.Elm/ElmServiceCollectionExtensions.cs]: > An `<see cref="Action{ElmOptions}"/>` to configure the provided `<see cref="ElmOptions"/>`.

**Comment by pranavkm** [comment]
:shipit: I'd run it by @HaoK's since his comment seems to indicate you'd have to redo these changes again.

**Comment by HaoK** [comment]
Yes IOptions is going away, so we definitely do not want to switch these middlewares over to using it when they aren't using them today

**Comment by HaoK** [comment]
@javiercn We decided against removing IOptions so you are free to progress now... thanks for waiting

File: src/Microsoft.AspNetCore.Diagnostics.Elm/ElmServiceCollectionExtensions.cs (hunk 2 above)
Comment (Eilon) [inline src/Microsoft.AspNetCore.Diagnostics.Elm/ElmServiceCollectionExtensions.cs]: I would spell it out as `Error Logging Middleware`

**Pull request merged**